Prox/Prox/Models/Event.swift

```diff
@@ -172,12 +172,22 @@ class Event {
 
     internal func shouldShowEvent(withStartTime startTime: Date, endTime: Date?, timeIntervalBeforeStartOfEvent startTimeInterval: TimeInterval, timeIntervalBeforeEndOfEvent endTimeInterval: TimeInterval, atCurrentTime currentTime: Date) -> Bool {
         if isFutureEvent(currentTime: currentTime) {
-            return (startTime - startTimeInterval) <= currentTime
+            let validFutureEvent = (startTime - startTimeInterval) <= currentTime
+            if !validFutureEvent {
+                NSLog("\(description) is a future event but (\(startTime) - \(startTimeInterval)) <= \(currentTime)")
```

Comment:
nit: I'd prefer not to duplicate logic in logs – the logic can change slightly and the comment could become misleading! I'd prefer to describe intent, which changes less often than logic and list the params, e.g.:

`NSLog("\(description) is a future event but is not valid. startTime: \(startTime); startTimeInterval: \(startTimeinterval); currentTime: \(currentTime)")`

Here and below.